src/content/tutorials/drizzle-with-db/drizzle-with-neon.mdx

```diff
@@ -0,0 +1,241 @@
+
+#### Retrieve your Neon database connection string
+
+Navigate to the **Connection Details** section in the project console to find your database connection string. It should look similar to this:
+
+```bash
+postgres://username:password@ep-cool-darkness-123456.us-east-2.aws.neon.tech/neondb
+```
+
+Add the `DATABASE_URL` environment variable to your `.env` file, which you'll use to connect to our Neon database.
+
+```bash
+DATABASE_URL=NEON_DATABASE_CONNECTION_STRING
+```
+
+#### Connect Drizzle ORM to your database 
+
+Create a `db.ts` file and set up your database configuration:
+
+```typescript copy filename="src/db.ts"
+import { drizzle } from "drizzle-orm/neon-http";
+import { neon } from "@neondatabase/serverless";
+import { config } from "dotenv";
+
+config({ path: ".env" });
+
+const sql = neon(process.env.DATABASE_URL!);
+export const db = drizzle(sql);
+```
+
+#### Create tables
+
+Create a `schema.ts` file and declare your tables:
+
+```typescript copy filename="src/schema.ts"
+import { pgTable, integer, serial, text, timestamp } from "drizzle-orm/pg-core";
+
+export const authors = pgTable("authors", {
+  id: serial("id").primaryKey(),
+  name: text("name").notNull(),
+  bio: text("bio"),
+  createdAt: timestamp("created_at").notNull().defaultNow(),
+});
+
+export const books = pgTable("books", {
+  id: serial("id").primaryKey(),
+  title: text("title").notNull(),
+  authorId: integer("author_id").references(() => authors.id),
+  createdAt: timestamp("created_at").notNull().defaultNow(),
+});
+```
+
+The code defines two tables: `authors`, which will contain the list of all the authors, and `books`, which will contain the list of books written by the authors. Each book is associated with an author using the `authorId` field.
+
+#### Generate migrations
+
+We'll use the drizzle-kit CLI tool. Run the following command in your terminal. 
+
+```bash
+npx drizzle-kit generate:pg --schema src/schema.ts --out ./drizzle
+```
+
+These migrations are stored in the `drizzle`  directory, as specified above. This directory will contain the SQL files necessary to update your database schema and a `meta` folder for storing snapshots of the schema at different migration stages. 
+
+Example of the generated migration:
+
+```sql
+CREATE TABLE IF NOT EXISTS "authors" (
+	"id" serial PRIMARY KEY NOT NULL,
+	"name" text NOT NULL,
+	"bio" text,
+	"created_at" timestamp DEFAULT now() NOT NULL
+);
+--> statement-breakpoint
+CREATE TABLE IF NOT EXISTS "books" (
+	"id" serial PRIMARY KEY NOT NULL,
+	"title" text NOT NULL,
+	"author_id" integer,
+	"created_at" timestamp DEFAULT now() NOT NULL
+);
+--> statement-breakpoint
+DO $$ BEGIN
+ ALTER TABLE "books" ADD CONSTRAINT "books_author_id_authors_id_fk" FOREIGN KEY ("author_id") REFERENCES "authors"("id") ON DELETE no action ON UPDATE no action;
+EXCEPTION
+ WHEN duplicate_object THEN null;
+END $$;
+```
+
+#### Applying changes to the database
+
+Create a `migrate.ts` file and add the code below. 
```

Comment:
There is `drizzle-kit migrate` command now to apply migrations. 

cc: https://orm.drizzle.team/kit-docs/upgrade-21